lib/functions.sh

```diff
@@ -907,6 +907,41 @@ make_install_in() {
         logerr "------ Make install in $1 failed"
 }
 
+make_lintlibs() {
+    logmsg "Making lint libraries"
+
+    LINTLIB=$1
+    LINTLIBDIR=$2
+    LINTINCDIR=$3
+    LINTINCFILES=$4
+
+    [[ -z ${LINTLIB} ]] && logerr "not lint library specified"
+    [[ -z ${LINTINCFILES} ]] && LINTINCFILES="*.h"
+
+    cat <<EOF > ${DTMPDIR}/${PKGD}_llib-l${LINTLIB}
+/* LINTLIBRARY */
+/* PROTOLIB1 */
+#include <sys/types.h>
+#undef _LARGEFILE_SOURCE
+EOF
+    pushd ${DESTDIR}${LINTINCDIR} > /dev/null
+	sh -c "eval /usr/gnu/bin/ls -U ${LINTINCFILES}" | \
```

Comment:
On Mon, May 04, 2015 at 09:43:55AM -0700, Albert Lee wrote:
> > +    LINTLIB=$1
> > +    LINTLIBDIR=$2
> > +    LINTINCDIR=$3
> > +    LINTINCFILES=$4
> > +
> > +    [[ -z ${LINTLIB} ]] && logerr "not lint library specified"
> > +    [[ -z ${LINTINCFILES} ]] && LINTINCFILES="*.h"
> > +
> > +    cat <<EOF > ${DTMPDIR}/${PKGD}_llib-l${LINTLIB}
> > +/* LINTLIBRARY */
> > +/* PROTOLIB1 */
> > +#include <sys/types.h>
> > +#undef _LARGEFILE_SOURCE
> > +EOF
> > +    pushd ${DESTDIR}${LINTINCDIR} > /dev/null
> > +	sh -c "eval /usr/gnu/bin/ls -U ${LINTINCFILES}" | \
> 
> I know we have a few other references to /usr/gnu/bin in the build
> infrastructure, but is there any reason to prefer this over simply
> `echo ${LINTINCFILES}`, given our /bin/sh is certainly capable of
> expanding the expressions?

The only reason I used /usr/gnu/bin/ls was that our ls -U had a bug back
than. That was https://www.illumos.org/issues/5613 and has since been
fixed.

I think the reason I used ls instead of echo is that I wanted one header
per line for the following sed expression to work. That probably can
also be achieved with echo and a different sed expression, but ls seemed
a natural way to do it.


Hans


-- 
%SYSTEM-F-ANARCHISM, The operating system has been overthrown